src/normalizeEntityUri.ts

```diff
@@ -51,20 +51,19 @@ function normalizeUri (uri: unknown, baseUrl: string): string | null {
   if (typeof uri !== 'string') return null
   const sorted = sortQueryParams(uri)
   const simpleReplace = sorted.replace(new RegExp(`^${baseUrl}`), '')
-  if (baseUrl && simpleReplace === uri) {
-    try {
-      const parsedBaseUrl = new URL(baseUrl)
-      const uriHasHost = getHostOfUri(uri) !== undefined
-      if (parsedBaseUrl.host && uriHasHost) {
-        return simpleReplace
-      }
 
-      const pathname = parsedBaseUrl.pathname.replace(/\/$/, '')
-      return sorted.replace(new RegExp(`^${pathname}`), '')
-    } catch (_) {
+  try {
+    const parsedBaseUrl = new URL(baseUrl)
+    const uriHasHost = getHostOfUri(uri) !== undefined
+    if (parsedBaseUrl.host && uriHasHost) {
+      return simpleReplace
     }
+
+    const pathname = parsedBaseUrl.pathname.replace(/\/$/, '')
+    return sorted.replace(new RegExp(`^${pathname}`), '')
```

Comment:
This could be considered a regexp vulnerability, because depending on the application, user controllable input may used as a regexp. A check with `startsWith` plus a `substr` should be a little safer here